hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/security/UserGroupInformation.java

```diff
@@ -237,15 +227,13 @@ public boolean commit() throws LoginException {
         } catch (Exception e) {
           throw (LoginException)(new LoginException(e.toString()).initCause(e));
         }
-        if (LOG.isDebugEnabled()) {
-          LOG.debug("User entry: \"" + userEntry.toString() + "\"" );
-        }
+        LOG.debug("User entry: \"{}\"", userEntry);
 
         subject.getPrincipals().add(userEntry);
         return true;
       }
-      LOG.error("Can't find user in " + subject);
-      throw new LoginException("Can't find user name");
+      LOG.error("Failed to find user name in " + subject);
```

Comment:
Use {}?

Reply:
Using a parameterized message is typically reserved for DEBUG and TRACE messages.  It is however, also helpful for readability in my opinion.  However, it's not that important here because ERROR level messages will always be printed, so any kind of optimization is futile.

However, you do bring to my attention that this is an example of the throw-and-log situation.  I changed to remove the logging altogether.